### PR TITLE
Modified to support ECDSA SSH keys.

### PR DESCRIPTION
**Pull request by cgruver**
**Fixes:** https://github.com/code-ready/crc/issues/1747

**Dependent on:** https://github.com/code-ready/snc/pull/289

Modified CRC to support the more secure ECDSA SSH cipher.

### CODE REVIEW

**Comment by guillaumerose** [comment]
/ok-to-test

Should we change also this in the installer ? snc will also add a RSA key in the bundle if I'm correct.

**Comment by guillaumerose** [comment]
If you can find a way to not break upgrade. It's maybe hard to do something backward compatible and not worth it.

**Comment by cgruver** [comment]
D'oh!   Lint check failed... I'll fix that.

**Comment by cgruver** [comment]
> If you can find a way to not break upgrade. It's maybe hard to do something backward compatible and not worth it.

Let's chat about that on Slack.  I'm not sure what it will take, but would be willing to take a crack at it.

**Review by gbraad — APPROVED** [review]

**Comment by gbraad** [comment]
backward compat is not really worth it.

perhaps detecting it and suggesting to recreate the machine. we are ephemeral anyway.

**Review by cfergeau — COMMENTED** [review]
The commit log is missing an explanation why we are doing that. It should document that this is due to the crypto policy changes in f33/fcos33 which reject connection with rsa keys, which we were using so far.

**Comment by cfergeau** [comment]
Can't backward compat be dealt with by this change in machine/machine.go ?
```
- sshRunner, err := crcssh.CreateRunner(instanceIP, getSSHPort(client.useVSock()), crcBundleMetadata.GetSSHKeyPath(), constants.GetPrivateKeyPath())
+ sshRunner, err := crcssh.CreateRunner(instanceIP, getSSHPort(client.useVSock()), crcBundleMetadata.GetSSHKeyPath(), constants.GetPrivateKeyPath(), crcBundleMetadata.GetOldSSHKeyPath())
```
(`GetOldSSHKeyPath()` will return `id_rsa.pub`)

**Comment by praveenkumar** [comment]
I don't understand why it is not the backward compatible (even CI is passing which is using the older bundles), initial ssh key information is part of bundle metadata, as long as it have correct details it will start the first ssh connection, now we generate the keypair (ecdsa) which is also valid for old bundles so, both the way it is going to work.

**Comment by guillaumerose** [comment]
If you start an instance with crc 1.20.0 and try to do crc status with this one, you will get an error because the ssh key is not found.

**Comment by praveenkumar** [comment]
Ah you mean for already existing instance (running/stopped)  not the one which you created with this changes either older or newer bundles.

**Comment by cgruver** [comment]
> Ah you mean for already existing instance (running/stopped) not the one which you created with this changes either older or newer bundles.

I get it, it's simply because of the file name change...  id_rsa to id_ecdsa.

**Comment by cgruver** [comment]
I think @cfergeau was correct.  Adding backward compatibility was pretty easy...  provided this worked.

File: pkg/crc/constants/constants.go [hunk 1]
```diff
@@ -144,10 +144,15 @@ func contains(arr []string, str string) bool {
...
-	return filepath.Join(MachineInstanceDir, DefaultName, "id_rsa.pub")
+	return filepath.Join(MachineInstanceDir, DefaultName, "id_ecdsa.pub")
 }
 
 func GetPrivateKeyPath() string {
+	return filepath.Join(MachineInstanceDir, DefaultName, "id_ecdsa")
+}
+
+// For backward compatibility to v 1.20.0
+func GetRsaPrivateKeyPath() string {
```
Comment (guillaumerose) [inline pkg/crc/constants/constants.go:155]: RSA is an acronym, so use capital letters? `GetRSAPrivateKeyPath`

File: pkg/crc/ssh/keys.go [hunk 2]
```diff
@@ -26,17 +27,17 @@ type KeyPair struct {
 // NewKeyPair generates a new SSH keypair
 // This will return a private & public key encoded as DER.
 func NewKeyPair() (keyPair *KeyPair, err error) {
-	priv, err := rsa.GenerateKey(rand.Reader, 2048)
+
```
Comment (guillaumerose) [inline pkg/crc/ssh/keys.go:30]: extra space

**Review by guillaumerose — APPROVED** [review]
2 nits but otherwise lgtm.

**Pull request merged**